tianshou/highlevel/experiment.py

```diff
@@ -204,43 +242,39 @@ def save(self, directory: str) -> None:
         logger_run_id: str | None = None,
         raise_error_on_dirname_collision: bool = True,
-        **kwargs: dict[str, Any],
-    ) -> ExperimentResult:
-        """Run the experiment and return the results.
-
-        :param run_name: Defines a name for this run of the experiment, which determines
-            the subdirectory (within the persistence base directory) where all results will be saved.
-            If None, the experiment's name will be used.
-            The name may contain path separators (i.e. `os.path.sep`, as used by `os.path.join`), in which case
-            a nested directory structure will be created.
-        :param logger_run_id: Run identifier to use for logger initialization/resumption (applies when
-            using wandb, in particular).
-        :param raise_error_on_dirname_collision: set to `False` e.g., when continuing a previously executed
-            experiment with the same name.
-        :param kwargs: for backward compatibility with old parameter names only
-        :return:
+        reset_collectors: bool = True,
+    ) -> World:
+        """Creates the world object for the experiment.
+
+        The world object contains all relevant instances for the experiment,
+        such as environments, policy, collectors, etc.
+        This method is the main entrypoint for users who don't want to use `run` directly. A common use case
+        is that some configuration or custom logic should happen before the training loop starts, but one
+        still wants to use the convenience of high-level interfaces for setting up the experiment.
+
+        :param override_experiment_name: whether to override the experiment name in the resulting `World`.
+            Passing `DATETIME_TAG` will use a name containing the current date and time.
+        :param logger_run_id: Run identifier to use for logger initialization/resumption.
+        :param raise_error_on_dirname_collision: whether to raise an error on collisions when creating the
+            persistence directory. Only takes effect if persistence is enabled. Set to `False` e.g., when continuing
+            a previously executed experiment with the same `persistence_base_dir` and name.
+        :param reset_collectors: whether to reset the collectors before training starts.
+            Setting to `False` can be useful when continuing training from a previous run with restored collectors,
+            or for adding custom logic before training starts.
         """
-        # backward compatibility
-        _experiment_name = kwargs.pop("experiment_name", None)
-        if _experiment_name is not None:
-            run_name = cast(str, _experiment_name)
-            deprecation(
-                "Parameter run_name should now be used instead of experiment_name. "
-                "Support for experiment_name will be removed in the future.",
-            )
-        assert len(kwargs) == 0, f"Received unexpected arguments: {kwargs}"
```

Comment:
Why was backward compatibility for parameter `experiment_name` of method `run` dropped?
Surely we want to remain compatible with v1.0.0, don't we?

Reply:
This change basically undoes changes/improvements from commit 35779696ee860684d1e0dfa229cc8f367ef5118f.

See `git difftool 35779696~1 35779696 tianshou/highlevel/experiment.py`.

Reply:
See below, was a copy-paste error. I didn't put enough attention on the deleted lines in the diff, thx for spotting!

Reply:
Done